src/arwave.py

```diff
@@ -1,122 +1,214 @@
+            np.median(arx_res), np.median(obj_res)))
+
+        #Determine sigma of gaussian for smoothing
+        arx_sig2 = (arx_disp*arx_wid[arx_w][arx_keep])**2.
+        obj_sig2 = (obj_disp*obj_wid[obj_w][obj_keep])**2.
+
+        arx_med_sig2 = np.median(arx_sig2)
+        obj_med_sig2 = np.median(obj_sig2)
+
+        if obj_med_sig2 >= arx_med_sig2:
+            smooth_sig = np.sqrt(obj_med_sig2-arx_med_sig2)  # Ang
+            smooth_sig_pix = smooth_sig / arx_disp
+        else:
+            msgs.warn("Prefer archival sky spectrum to have higher resolution")
+            smooth_sig_pix = 0.
+            #smooth_sig = np.sqrt(arx_med_sig**2-obj_med_sig**2)
+
+        #Determine region of wavelength overlap
+        min_wave = max(np.amin(arx_sky.wavelength.value), np.amin(obj_sky.wavelength.value))
+        max_wave = min(np.amax(arx_sky.wavelength.value), np.amax(obj_sky.wavelength.value))
+
+        #Smooth higher resolution spectrum by smooth_sig (flux is conserved!)
+        if np.median(obj_res) >= np.median(arx_res):
+            msgs.warn("New Sky has higher resolution than Archive.  Not smoothing")
+            #obj_sky_newflux = ndimage.gaussian_filter(obj_sky.flux, smooth_sig)
+        else:
+            #tmp = ndimage.gaussian_filter(arx_sky.flux, smooth_sig)
+            arx_sky = arx_sky.gauss_smooth(smooth_sig_pix*2.355)
+            #arx_sky.flux = ndimage.gaussian_filter(arx_sky.flux, smooth_sig)
+
+        # Define wavelengths of overlapping spectra
+        keep_idx = np.where((obj_sky.wavelength.value>=min_wave) &
+                             (obj_sky.wavelength.value<=max_wave))[0]
+        #keep_wave = [i for i in obj_sky.wavelength.value if i>=min_wave if i<=max_wave]
+
+        #Rebin both spectra onto overlapped wavelength range
+        if len(keep_idx) <= 50:
+            msgs.error("Not enough overlap between sky spectra")
+
+        else: #rebin onto object ALWAYS
+            keep_wave = obj_sky.wavelength[keep_idx]
+            arx_sky = arx_sky.rebin(keep_wave)
+            obj_sky = obj_sky.rebin(keep_wave)
+
+
+        #deal with bad pixels
+        msgs.work("Need to mask bad pixels")
+
+        #deal with underlying continuum
+        msgs.work("Need to deal with underlying continuum")
+
+        #Cross correlation of spectra
+        corr = np.correlate(arx_sky.flux, obj_sky.flux, "same")
+
+        #Create array around the max of the correlation function for fitting for subpixel max
+        # Restrict to pixels within max_shift of zero lag
+        lag0 = corr.size/2
+        mxshft = slf._argflag['reduce']['flexure']['max_shift']
+        max_corr = np.argmax(corr[lag0-mxshft:lag0+mxshft]) + lag0-mxshft
+        subpix_grid = np.linspace(max_corr-3., max_corr+3., 7.)
+
+        #Fit a 2-degree polynomial to peak of correlation function
+        fit = np.polynomial.polynomial.polyfit(subpix_grid, corr[subpix_grid.astype(np.int)], 2)
+        roots = np.roots([fit[2], fit[1], fit[0]])
```

Comment:
Using `func_fit`, you may still have to reverse the array (but I'm not 100% sure). A cleaner form would be:
`roots = np.roots(fit[::-1])`